go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy_controller.go

```diff
@@ -0,0 +1,509 @@
+	}
+
+	klog.Info("Setting up event handlers for Namespaces in Admin Network Policy controller")
+	c.anpNamespaceLister = namespaceInformer.Lister()
+	c.anpNamespaceSynced = namespaceInformer.Informer().HasSynced
+	c.anpNamespaceQueue = workqueue.NewNamedRateLimitingQueue(
+		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
+		"anpNamespaces",
+	)
+	_, err = namespaceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.onANPNamespaceAdd,
+		UpdateFunc: c.onANPNamespaceUpdate,
+		DeleteFunc: c.onANPNamespaceDelete,
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("could not add Event Handler for namespace Informer during admin network policy controller initialization, %w", err)
+	}
+
+	klog.Info("Setting up event handlers for Pods in Admin Network Policy controller")
+	c.anpPodLister = podInformer.Lister()
+	c.anpPodSynced = podInformer.Informer().HasSynced
+	c.anpPodQueue = workqueue.NewNamedRateLimitingQueue(
+		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
+		"anpPods",
+	)
+	_, err = podInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.onANPPodAdd,
+		UpdateFunc: c.onANPPodUpdate,
+		DeleteFunc: c.onANPPodDelete,
+	}))
+	if err != nil {
+		return nil, fmt.Errorf("could not add Event Handler for pod Informer during admin network policy controller initialization, %w", err)
+	}
+
+	// TODO(tssurya): We don't use recorder now but will add events in future iterations
+	c.eventRecorder = recorder
+
+	return c, nil
+}
+
+// Run will not return until stopCh is closed. workers determines how many
+// objects (pods, namespaces, anps, banps) will be handled in parallel.
+func (c *Controller) Run(threadiness int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+
+	klog.Infof("Starting controller %s", c.controllerName)
+
+	// Wait for the caches to be synced
+	klog.Info("Waiting for informer caches to sync")
+	if !util.WaitForNamedCacheSyncWithTimeout(c.controllerName, stopCh, c.anpCacheSynced, c.banpCacheSynced, c.anpNamespaceSynced, c.anpPodSynced) {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for caches to sync"))
+		klog.Errorf("Error syncing caches for admin network policy and baseline admin network policy")
+		return
+	}
+
+	klog.Infof("Repairing Admin Network Policies")
+	// Run the repair function at startup so that we synchronize KAPI and OVNDBs
+	err := c.repairAdminNetworkPolicies()
+	if err != nil {
+		klog.Errorf("Failed to repair Admin Network Policies: %v", err)
+	}
+	err = c.repairBaselineAdminNetworkPolicy()
+	if err != nil {
+		klog.Errorf("Failed to repair Baseline Admin Network Policy: %v", err)
+	}
+
+	wg := &sync.WaitGroup{}
+	// Start the workers after the repair loop to avoid races
+	klog.Info("Starting Admin Network Policy workers")
+	for i := 0; i < threadiness; i++ {
```

Comment:
I am thinking now if it makes sense to use `threadiness` here, when we know we have a centralized lock for every handler? it's not dangerous in any way, just a bit misleading...

Reply:
i get this confusion, even if we want to enable multi threading we still allow for a single thread only to be executing at a given time. this is again something that I have written in the TODOs: to see if we want to improve this after doing scale testing to allow for smaller locks.